tests/functional/UseCase/Table/ExportTableToFileTest.php

```diff
@@ -0,0 +1,483 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StorageDriver\FunctionalTests\UseCase\Table;
+
+use Doctrine\DBAL\Connection;
+use Generator;
+use Google\Protobuf\Any;
+use Google\Protobuf\Internal\GPBType;
+use Google\Protobuf\Internal\RepeatedField;
+use Keboola\CsvOptions\CsvOptions;
+use Keboola\StorageDriver\Command\Bucket\CreateBucketResponse;
+use Keboola\StorageDriver\Command\Table\ImportExportShared;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\ExportOptions;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\FileFormat;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\FilePath;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\FileProvider;
+use Keboola\StorageDriver\Command\Table\ImportExportShared\S3Credentials;
+use Keboola\StorageDriver\Command\Table\TableExportToFileCommand;
+use Keboola\StorageDriver\Command\Table\TableImportFromFileCommand;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\FunctionalTests\BaseCase;
+use Keboola\StorageDriver\Teradata\Handler\Table\Export\ExportTableToFileHandler;
+use Keboola\StorageDriver\Teradata\Handler\Table\Import\ImportTableFromFileHandler;
+use Keboola\TableBackendUtils\Column\ColumnCollection;
+use Keboola\TableBackendUtils\Column\Teradata\TeradataColumn;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Keboola\TableBackendUtils\Table\Teradata\TeradataTableDefinition;
+use Keboola\TableBackendUtils\Table\Teradata\TeradataTableQueryBuilder;
+
+class ExportTableToFileTest extends BaseCase
+{
+    protected GenericBackendCredentials $projectCredentials;
+
+    protected CreateBucketResponse $bucketResponse;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->cleanTestProject();
+
+        [$projectCredentials, $projectResponse] = $this->createTestProject();
+        $this->projectCredentials = $projectCredentials;
+
+        [$bucketResponse,] = $this->createTestBucket($projectCredentials, $projectResponse);
+        $this->bucketResponse = $bucketResponse;
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+        $this->cleanTestProject();
+    }
+
+    /**
+     * @dataProvider simpleExportProvider
+     */
+    public function testExportTableToFile(bool $isCompressed, int $exportSize): void
+    {
+        $bucketDatabaseName = $this->bucketResponse->getCreateBucketObjectName();
+        $sourceTableName = md5($this->getName()) . '_Test_table_export';
+        $exportDir = sprintf(
+            'export/%s/',
+            str_replace([' ', '"', '\''], ['-', '_', '_'], $this->getName())
```

Comment:
takové sraní :D já bych tam hodil md5 :D

Reply:
tak lip se to cte, ale OK :)

Reply:
klidně to nech tak jak to máš, jen sem si říkal, že by sa mě nechtělo zjišťovat co všecko v tom name može byt za bordel a dělat replace :D

Reply:
no prave, jsem se nad tim znova zamyslel a radsi jsem to zmenil 🤷